spec/support/sentry_for_rspec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module SentryForRSpec
+  def self.report_retry(example, config)
+    create_sentry_scope(example, config)
+
+    Sentry.with_scope do |scope|
+      create_tags(scope, example)
+      send_message(example)
```

Comment:
send the message within the scope so it picks up the tag. A message sent outside the scope loop won't apply the tag.